Project7.md

```diff
@@ -172,12 +178,61 @@ It is also very useful to write comments for each line describing what the forma
+
+```
+item_id  | recommended_items
+---------|------------------
+1        |  5, 7, 9,
+2        |  3, 4
+3        |  4, 2, 1, 5,
+```
+
+
+To populate and access this recommendations table, you will interface with the MySQL database directly with a [python MySQL client](https://github.com/PyMySQL/mysqlclient-python) rather than Django's ORM. Population of the table will occur during the Pyspark job. Within your Pyspark program, you will need to:
+
+1. Initiate a connection with the database using PyMySQL.
+2. Create a recommendations table with the appropriate column types if there is not already one, and insure sure it is empty.
+3. Make appropriate INSERT queries while iterating through co-view results after the ```collect``` call, making sure to account for items that already have rows in the table, as well as those that don't.
+
+The spark containers that will contain your Pyspark program don't come with everything needed for PyMySQL right off the bat, so some package installation must be taken care of. After 'docker-composing up' but before running your spark job for the first time, manually enter the two spark containers and run the following commands (either individually by hand, or in script form like below):
+
+```
+#!/usr/bin/env bash
+apt-get update &&
+apt-get install python3-dev libmysqlclient-dev -y &&
+apt-get install python-pip -y &&
+pip install mysqlclient &&
+apt-get install python-mysqldb
+```
+
+You should now be ready to get your hands dirty with some raw SQL queries! There's plenty of [helpful documentation](https://www.tutorialspoint.com/python/python_database_access.htm) all over the web to get you started, but here are some tips that may be of use:
+* To see what your SQL queries are actually doing to the database, you can access MySQL through the mysql-cmdline container and see what's exactly in the recommendations table with the commands ```USE cs4501;``` and ```SELECT * FROM recommendations;```
+* Some SQL syntax you might want to consider looking into is ```ON DUPLICATE KEY UPDATE```, ```CONCAT()```, and ```TRUNCATE```.
+* With all the containers you have running at this point, Spark may not have enough RAM to work properly. The errors aren't too helpful when this happens, so just be aware when facing obscure problems that you may need to just increase the amount of RAM your computer allocates to Docker.
+
+Once you have correctly populated the database with Sparks output for the recommendation service, you can now move on to implementing the service itself in the experience layer. In every item detail page, you must now display a few of it's recommendations if it has any. You will also need to use PyMySQL for this, but only to read from the database table. Note that you may need to install some of the same packages that you needed in the Spark containers in the experience container as well.
```

Comment:
Perhaps better to say create a Django model for this table as well so that your models layer and experience service layer can continue to use the Django ORM?